### DISCUSSION TITLE
Risks of using .coreAnimation with assert suppressed

### DISCUSSION THREAD

**Discussion by iq3addLi** [airbnb/lottie-ios#1729] [Q&A] [answered]
Hi there!

I'm enjoying lottie-ios!
I was having a problem with the lottie animations not displaying as intended.
By chance I found the following [comment](https://github.com/airbnb/lottie-ios/issues/1648#issuecomment-1181060410), so I suppressed assert and used the `.coreAnimation` engine.
Although there were a few problems, I'm able to obtain an animation close to what I had envisioned.

So my question is, what are the **risks** of using the coreAnimation engine with Assert suppressed?
What I can assume is that I'm not aware of unsupported features being included.
Are there any serious problems? such as broken backward compatibility.

Thank you for your reading🙏.

**Comment by calda** [comment, marked as answer]
The Core Animation rendering engine emits assertions if it detects that it is unable to correctly render an animation. Disabling the assertions just means that you won't be alerted when using an unsupported animation / feature -- it would be up to you to check whether the animation plays correctly or not. 

imo disabling assertions within a specific `AnimationView` (e.g. `AnimationView(animation: myAnimation, logger: .printToConsole)`)  that always displays a specific animation, which you have already verified animates correctly, is very safe and has no real downsides. 

Disabling assertions globally in your app (e.g. `LottieLogger.shared = .printToConsole`) is a bit more risky if you have a large number of Lottie animations and haven't manually verified that all of them animate correctly. Of course, the worst thing that can happen is only that an animation appears slightly incorrectly.

**Comment by iq3addLi** [comment]
@calda 
Now I understand what I need to pay attention to.
In lottie-ios 3.4.2, we have found almost no problems with playback. I was using 3.4.1 at the time of the question.
Therefore, I'm going to use the coreAnimation engine with assert suppressed.

There is still a warning about `time remapping keyframes` as to why they need to be suppressed, however I will start a separate thread of questions on this.

Thank you for your informative response!
